src/runtime-helpers/runtime-helpers.ts

```diff
@@ -213,6 +227,37 @@ module ScriptLab {
     function _isDialogApiSupported() {
         return (window as any).Office.context.requirements.isSetSupported('DialogApi');
     }
+
+    function _extractAndCacheAccessToken(message: string, clientId: string, resource: string): string {
+        const authResponseKeyValues = message.split('&');
```

Comment:
Please change to do JSON-parsing instead, since you'll be sending a JSON-stingified object (see comment above).  You can see why it's useful, too:  already this code is starting to make weird assumptions about the order of parameters, etc.  I was being lazy last time while it was only one parameter, though I probably should have known better even then.  But now that it's more than one parameter, it's time to refactor.